src/libraries/Microsoft.XmlSerializer.Generator/tests/SGenTests.cs

```diff
@@ -31,6 +31,9 @@ public static void SgenCommandTest()
 
             const string CodeFile = "SerializableAssembly.XmlSerializers.cs";
             const string LKGCodeFile = "Expected.SerializableAssembly.XmlSerializers.cs";
+
+            File.WriteAllText(LKGCodeFile, LineEndingsHelper.Normalize(File.ReadAllText(LKGCodeFile)));
```

Comment:
this will change the file inside the tree -- it will show up as a git edit to anyone that runs the test.

Instead you'd write it to temp like this: https://github.com/danmoseley/runtime/blob/19eb98df125a6bc766acdb8aacfc36c7d2a75403/src/libraries/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs#L3914

But you don't need it in a file, right? You just need the length of `LineEndingsHelper.Normalize(File.ReadAllText(LKGCodeFile))`

Reply:
Yeah, you are right indeed. Also this PR isn't complete, I didn't mean to open it here, I wanted to open it to my own fork so I could track the overall change

Reply:
Relatively small PR's (that stand on their own) are usually better than large ones, especially for contributors that are still getting used to the repo. They're faster to review, and if there's a common mistake, it doesn't have to be flagged/fixed in so many places. Large PR's are so hard to grok that sometimes they sit for some time. So I suggest to not make this PR bigger.

Reply:
@danmoseley Ok, this PR already provides fixes for 4 of the tests, once this is closed I'll get working on the rest